http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -0,0 +1,194 @@
+package org.broadinstitute.dsde.workbench.leonardo.monitor
+
+import cats.effect.{Async, Concurrent, ContextShift, IO, Timer}
+import org.broadinstitute.dsde.workbench.google2.{Event, GoogleSubscriber}
+import fs2.{Pipe, Stream}
+import io.circe.{Decoder, DecodingFailure, Encoder}
+import org.broadinstitute.dsde.workbench.leonardo.MachineConfig
+import org.broadinstitute.dsde.workbench.leonardo.JsonCodec._
+import org.broadinstitute.dsde.workbench.leonardo.db.{clusterQuery, followupQuery, DbReference}
+import org.broadinstitute.dsde.workbench.leonardo.model.google.{ClusterStatus, MachineType}
+import org.broadinstitute.dsde.workbench.leonardo.util.ClusterHelper
+import _root_.io.chrisdavenport.log4cats.Logger
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.leonardo.model.LeoException
+import org.broadinstitute.dsde.workbench.leonardo.model.google.ClusterStatus.Stopped
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage._
+import org.broadinstitute.dsde.workbench.model.WorkbenchException
+import cats.implicits._
+
+import scala.concurrent.ExecutionContext
+
+case class ClusterNotFoundException(
+  clusterId: Long,
+  override val message: String =
+    "Could not process ClusterTransitionFinishedMessage because it was not found in the database"
+) extends LeoException
```

Comment:
When you extend `LeoException` you should provide a status otherwise it'll default to 500. For example: https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala#L51

Reply:
…although these errors are not returned directly to users, right? How do errors manifest? I'll keep reviewing & try to figure it out :)

Reply:
they shouldn't be presented to the user, they are printed in line 46 of the messageResponder via

```
response.onError {
      case e => IO(logger.error(s"Unable to process a message in received from pub/sub subscription in messageResponder: ${message}, errorMessage: ${e.getMessage}", e))
    }
```